core/src/main/java/com/netflix/conductor/core/dal/ExecutionDAOFacade.java

```diff
@@ -241,44 +248,57 @@ public long getPendingWorkflowCount(String workflowName) {
     /**
      * Creates a new workflow in the data store
      *
-     * @param workflow the workflow to be created
+     * @param workflowModel the workflow to be created
      * @return the id of the created workflow
      */
-    public String createWorkflow(WorkflowModel workflow) {
-        WorkflowModel leanWorkflow = modelMapper.getLeanCopy(workflow);
-        executionDAO.createWorkflow(leanWorkflow);
+    public String createWorkflow(WorkflowModel workflowModel) {
+        externalizeWorkflowData(workflowModel);
+        executionDAO.createWorkflow(workflowModel);
         // Add to decider queue
         queueDAO.push(
                 DECIDER_QUEUE,
-                leanWorkflow.getWorkflowId(),
-                leanWorkflow.getPriority(),
+                workflowModel.getWorkflowId(),
+                workflowModel.getPriority(),
                 properties.getWorkflowOffsetTimeout().getSeconds());
         if (properties.isAsyncIndexingEnabled()) {
-            indexDAO.asyncIndexWorkflow(new WorkflowSummary(modelMapper.getWorkflow(leanWorkflow)));
+            indexDAO.asyncIndexWorkflow(new WorkflowSummary(workflowModel.toWorkflow()));
         } else {
-            indexDAO.indexWorkflow(new WorkflowSummary(modelMapper.getWorkflow(leanWorkflow)));
+            indexDAO.indexWorkflow(new WorkflowSummary(workflowModel.toWorkflow()));
         }
-        return leanWorkflow.getWorkflowId();
+        return workflowModel.getWorkflowId();
+    }
+
+    private void externalizeTaskData(TaskModel taskModel) {
+        externalPayloadStorageUtils.verifyAndUpload(
+                taskModel, ExternalPayloadStorage.PayloadType.TASK_INPUT);
+        externalPayloadStorageUtils.verifyAndUpload(
+                taskModel, ExternalPayloadStorage.PayloadType.TASK_OUTPUT);
+    }
+
+    private void externalizeWorkflowData(WorkflowModel workflowModel) {
+        externalPayloadStorageUtils.verifyAndUpload(
+                workflowModel, ExternalPayloadStorage.PayloadType.WORKFLOW_INPUT);
+        externalPayloadStorageUtils.verifyAndUpload(
+                workflowModel, ExternalPayloadStorage.PayloadType.WORKFLOW_OUTPUT);
     }
 
     /**
      * Updates the given workflow in the data store
      *
-     * @param workflow the workflow tp be updated
+     * @param model the workflow tp be updated
      * @return the id of the updated workflow
      */
-    public String updateWorkflow(WorkflowModel workflow) {
-        workflow.setUpdatedTime(System.currentTimeMillis());
-        if (workflow.getStatus().isTerminal()) {
-            workflow.setEndTime(System.currentTimeMillis());
+    public String updateWorkflow(WorkflowModel model) {
```

Comment:
rename to `workflowModel` to make it consistent with others?

Reply:
done